common/client-libs/validator-client/src/nyxd/cosmwasm_client/client_traits/query_client.rs

```diff
@@ -218,17 +218,19 @@ pub trait CosmWasmClient: TendermintRpcClient {
 
         loop {
             let mut res = self
-                .tx_search(query.clone(), false, page, 100, Order::Ascending)
+                .tx_search(query.clone(), false, page, per_page, Order::Ascending)
                 .await?;
 
-            results.append(&mut res.txs);
             // sanity check for if tendermint's maximum per_page was modified -
             // we don't want to accidentally be stuck in an infinite loop
-            if res.total_count == 0 || res.txs.is_empty() {
+            let early_break = res.total_count == 0 || res.txs.is_empty();
+            results.append(&mut res.txs);
+
+            if early_break {
```

Comment:
Why instead of 
1. check
2. append
3. if (above check)

don't you just
1. append
2. check
```rust
results.append(&mut res.txs);

if res.total_count == 0 || res.txs.is_empty(); {
```

?

Reply:
because appending removes the data from the original vector and thus `res.txs.is_empty()` would always be true - which was the bug I was attempting to resolve : )